phy/cluster/manual/session.py

```diff
@@ -341,6 +343,96 @@ def on_cluster(self, up=None):
         self._store_extra_fields(up.added)
 
 
+class Waveforms(StoreItem):
+    """A cluster store item that manages the waveforms of all clusters."""
+    name = 'waveforms'
+    fields = [('waveforms', 'disk', np.float32,),
+              ('waveforms_spikes', 'disk', np.int64,),
+              ('mean_waveforms', 'memory'),
+              ]
+
+    def __init__(self, *args, **kwargs):
+        self._pr = kwargs.pop('progress_reporter')
+        self.n_spikes_max = kwargs.pop('n_spikes_max')
+        self.excerpt_size = kwargs.pop('excerpt_size')
+
+        super(Waveforms, self).__init__(*args, **kwargs)
+
+        self.n_channels = len(self.model.channel_order)
+        self.n_spikes = self.model.n_spikes
+        self.n_samples = self.model.n_samples_waveforms
+
+        # Set the shape of the features and masks.
+        self.fields[0] = ('waveforms', 'disk',
+                          np.float32, (-1, self.n_samples, self.n_channels))
+
+        self._selector = Selector(self.model.spike_clusters,
+                                  n_spikes_max=self.n_spikes_max,
+                                  excerpt_size=self.excerpt_size,
+                                  )
+
+    def store_cluster(self, cluster, spikes=None, mode=None):
+        spikes = self._selector.subset_spikes_clusters([cluster])
+        waveforms = self.model.waveforms[spikes]
+        self.disk_store.store(cluster,
+                              waveforms=waveforms.astype(np.float32),
+                              waveforms_spikes=spikes.astype(np.int64),
+                              )
+        self.memory_store.store(cluster,
+                                mean_waveforms=waveforms.mean(axis=0),
+                                )
+
+    def store_all_clusters(self, mode=None):
+        """Copy all data for that item from the model to the cluster store."""
+        clusters = self.to_generate(mode)
+        self._pr.value_max = len(clusters)
+        for cluster in clusters:
+            debug("Loading {0:s}, cluster {1:d}...".format(self.name,
+                  cluster))
+            self.store_cluster(cluster,
+                               spikes=self._spikes_per_cluster[cluster],
+                               mode=mode,
+                               )
+            self._pr.value += 1
+        self._pr.set_complete()
+
+    def is_consistent(self, cluster, spikes):
```

Comment:
for features I know the number of spikes per cluster so  I use that for consistency check
for waveforms not all spikes are stored. so I need to store them explicitely, and I use that for consistency